sklearnex/preview/ensemble/_forest.py

```diff
@@ -69,6 +70,100 @@
 
 
 class BaseForest(ABC):
+    _onedal_factory = None
+
+    def _onedal_fit(self, X, y, sample_weight=None, queue=None):
```

Comment:
Similarity between regression and classification unify to BaseForest. This simplifies maintenance and follows sklearn.